pkg/corerp/handlers/azure_nsg.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+	"github.com/radius-project/radius/pkg/azure/armauth"
+	rpv1 "github.com/radius-project/radius/pkg/rp/v1"
+	"github.com/radius-project/radius/pkg/to"
+)
+
+func NewAzureNSGHandler(arm *armauth.ArmConfig) ResourceHandler {
+	return &azureNSGHandler{arm: arm}
+}
+
+type azureNSGHandler struct {
+	arm *armauth.ArmConfig
+}
+
+func (handler *azureNSGHandler) Put(ctx context.Context, options *PutOptions) (map[string]string, error) {
+	nsg, ok := options.Resource.CreateResource.Data.(*armnetwork.SecurityGroup)
+	if !ok {
+		return nil, errors.New("cannot parse subnet")
+	}
+
+	subID := options.Resource.ID.FindScope("subscriptions")
+	resourceGroupName := options.Resource.ID.FindScope("resourceGroups")
+	if subID == "" || resourceGroupName == "" {
+		return nil, fmt.Errorf("cannot find subscription or resource group in resource ID %s", options.Resource.ID)
+	}
+
+	if options.Resource.LocalID == rpv1.LocalIDAzureAppGWNetworkSecurityGroup {
+		publicIP, ok := options.DependencyProperties[rpv1.LocalIDAzurePublicIP]["publicIPAddress"]
```

Comment:
If "publicIPAddress" does not exist this will panic. But OK we are guaranteed to have this value.